statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/StatefulFunctionsConfig.java

```diff
@@ -243,4 +246,34 @@ public void addAllGlobalConfigurations(Map<String, String> globalConfigurations)
   public void setGlobalConfiguration(String key, String value) {
     this.globalConfigurations.put(key, value);
   }
+
+  private static void validateStrictlyRequiredConfigs(Configuration configuration) {
+    final Set<String> parentFirstClassloaderPatterns =
+        parentFirstClassloaderPatterns(configuration);
+    if (!parentFirstClassloaderPatterns.contains("org.apache.flink.statefun")
+        && !parentFirstClassloaderPatterns.contains("org.apache.kafka")
```

Comment:
shouldn't that be an `||`instead of `&&` ?

Reply:
Yes you're right, it should be `||` :)

Reply:
nit: I think the semantics would be clearer if you put the strings in a static final List and then did `parentFirstClassloaderPatterns.containsAll`.